sandbox/libs/authn/src/main/java/org/opensearch/authn/PermissionHandler.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+import org.opensearch.authn.internal.InternalSubject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Objects;
+
+public class PermissionHandler {
+
+    /**
+     * Check that the permission required matches the permission available
+     */
+    public static boolean matchPermission(final Permission permission, final String permissionRequired) {
```

Comment:
This seems like a properly of a permissions grant, it can match a required permissions, why not have it on the permissions object?

Reply:
I separated this from the Permission object because I thought it would be best to keep any multi-permission functionality inside the PermissionHandler class. I felt that any instance of a permission should only be aware of itself and not be concerned with logic dealing with multiple permissions or their own correctness outside of syntax. If this function were part of the Permission class it would imply that the permissions were aware of a 'correct' alternative to themselves if they were malformed. 

I can swap this to be inside the Permissions class if this is preferred.